pkg/util/watcher/watch.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watcher
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+type ListWatchMetrics struct {
+	WatchTotal *prometheus.CounterVec
+	ListTotal  *prometheus.CounterVec
+}
+
+// NewListWatchMetrics takes in a prometheus registry and initializes
+// and registers the kube_state_metrics_list_total and kube_state_metrics_watch_total
+// metrics. It returns those registered metrics.
+func NewListWatchMetrics(r *prometheus.Registry) *ListWatchMetrics {
+	var m ListWatchMetrics
+	m.WatchTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "kube_state_metrics_watch_total",
+			Help: "Number of total resource watches in kube-state-metrics",
```

Comment:
Would this be better?
```suggestion
			Help: "Total number of resource watches attempted in kube-state-metrics",
```